editor/effects.js

```diff
@@ -19,9 +19,10 @@ export default {
 	REQUEST_POST_UPDATE( action, store ) {
 		const { dispatch } = store;
 		const { postId, edits } = action;
-		const isNew = ! postId;
 		const toSend = postId ? { id: postId, ...edits } : edits;
+		const isNew = ! postId;
 
+		dispatch( { type: 'CLEAR_POST_EDITS' } );
```

Comment:
In fact, I notice something odd about this behavior. When changing visibility for a published post and Update-ing the post, the visibility reverts back to its original value briefly. I suspect it's because of this dispatch here.